modules/contact/src/actions/ContactAction.C

```diff
@@ -901,6 +977,101 @@ ContactAction::addNodeFaceContact()
   }
 }
 
+void
+ContactAction::createSidesetPairsFromGeometry()
+{
+  mooseInfo("The contact action is reading the list of sidesets and matching them assuming that: "
+            "1) Those sidesets wrap around the surfaces of the bodies, 2) Axisymmetry of the "
+            "geometry and the mesh exists. Deviations from axisymmetry may give rise to wrong "
+            "contact pairing. The user is encouraged to check the numerical results.");
+
+  _mesh = &_problem->mesh();
```

Comment:
@lindsayad do we always get a serial mesh here? I am getting that the mesh is always serial despite using the `--distributed-mesh` option.

Reply:
Are you checking with `is_serial()`? At this stage of simulation setup, unless you've loaded a pre-split mesh, the mesh should indeed be serial. For any problems with displaced mesh, we delay remote element deletion until right before the `init_problem` task (which is close to the last task)

Reply:
If you're seeing that it's serial even during the simulation execution, then I would be pretty surprised. I would try rebasing on https://github.com/idaholab/moose/pull/24018 and see if that changes things

Reply:
Yes, checked with `is_serial()` and was only referring to this particular stage of system initialization. 

According to what you explained, if we wanted to make action decisions based on the mesh, it seems it'd be okay to do it at a stage such as the one in this PR and check that the mesh is serial to avoid issues with pre-split meshes (which are not being used much in contact problems at the moment)?

Reply:
Yes that sounds very reasonable to me